export_test.go

```diff
@@ -0,0 +1,327 @@
+					break
+				}
+				require.NoError(t, err)
+				err = importer.Add(item)
+				require.NoError(t, err)
+			}
+
+			require.Equal(t, tc.tree.Hash(), newTree.Hash(), "Tree hash mismatch")
+			require.Equal(t, tc.tree.Size(), newTree.Size(), "Tree size mismatch")
+			require.Equal(t, tc.tree.Version(), newTree.Version(), "Tree version mismatch")
+
+			tc.tree.Iterate(func(key, value []byte) bool {
+				index, _ := tc.tree.Get(key)
+				newIndex, newValue := newTree.Get(key)
+				require.Equal(t, index, newIndex, "Index mismatch for key %v", key)
+				require.Equal(t, value, newValue, "Value mismatch for key %v", key)
+				return false
+			})
+		})
+	}
+}
+
+func TestExporter_Close(t *testing.T) {
+	tree := setupExportTreeSized(t, 4096)
+	exporter := tree.Export()
+
+	node, err := exporter.Next()
+	require.NoError(t, err)
+	require.NotNil(t, node)
+
+	exporter.Close()
+	node, err = exporter.Next()
+	require.Error(t, err)
+	require.Equal(t, ExportDone, err)
+	require.Nil(t, node)
+
+	node, err = exporter.Next()
+	require.Error(t, err)
+	require.Equal(t, ExportDone, err)
+	require.Nil(t, node)
+
+	exporter.Close()
+	exporter.Close()
+}
+
+func TestExporter_DeleteVersionErrors(t *testing.T) {
+	tree, err := NewMutableTree(db.NewMemDB(), 0)
+	require.NoError(t, err)
+
+	tree.Set([]byte("a"), []byte{1})
+	_, _, err = tree.SaveVersion()
+	require.NoError(t, err)
+
+	tree.Set([]byte("b"), []byte{2})
+	_, _, err = tree.SaveVersion()
+	require.NoError(t, err)
+
+	tree.Set([]byte("c"), []byte{3})
+	_, _, err = tree.SaveVersion()
+	require.NoError(t, err)
+
+	itree, err := tree.GetImmutable(2)
+	require.NoError(t, err)
+	exporter := itree.Export()
+	defer exporter.Close()
+
+	err = tree.DeleteVersion(2)
+	require.Error(t, err)
+	err = tree.DeleteVersion(1)
+	require.NoError(t, err)
+
+	exporter.Close()
+	err = tree.DeleteVersion(2)
+	require.NoError(t, err)
+}
+
+func TestExporter_PruneVersionIgnores(t *testing.T) {
+	tree, err := NewMutableTreeWithOpts(db.NewMemDB(), db.NewMemDB(), 0, &Options{
+		KeepEvery:  10,
+		KeepRecent: 2,
+	})
+	require.NoError(t, err)
+
+	tree.Set([]byte("a"), []byte{1})
+	_, _, err = tree.SaveVersion()
+	require.NoError(t, err)
+
+	tree.Set([]byte("b"), []byte{2})
+	_, _, err = tree.SaveVersion()
+	require.NoError(t, err)
+
+	tree.Set([]byte("c"), []byte{3})
+	_, _, err = tree.SaveVersion()
+	require.NoError(t, err)
+
+	require.Equal(t, []int{2, 3}, tree.AvailableVersions())
+
+	itree, err := tree.GetImmutable(2)
+	require.NoError(t, err)
+	exporter := itree.Export()
+	defer exporter.Close()
+
+	tree.Set([]byte("d"), []byte{4})
+	_, _, err = tree.SaveVersion()
+	require.NoError(t, err)
+
+	require.Equal(t, []int{2, 3, 4}, tree.AvailableVersions())
+
+	exporter.Close()
+	tree.Set([]byte("e"), []byte{5})
+	_, _, err = tree.SaveVersion()
+	require.NoError(t, err)
+
+	// FIXME This seems like a bug in the version pruner; version 2 should have been pruned here
+	require.Equal(t, []int{2, 4, 5}, tree.AvailableVersions())
```

Comment:
@AdityaSripal Seems to be a bug in the tree pruner here - it only deletes the most recent version that is eligible for pruning, and does not check for any previous versions.